sdk/metric/doc.go

```diff
@@ -39,6 +39,30 @@
 // Meter.RegisterCallback and Registration.Unregister to add and remove
 // callbacks without leaking memory.
 //
+// # Cardinality Limits
+//
+// Cardinality refers to the number of unique attribute combinations for a single
+// metric within one collection (export) interval. High cardinality can lead to
+// excessive memory usage, increased storage costs, and backend performance issues.
+//
+// The Go OpenTelemetry SDK does not enforce a cardinality limit by default.
+// A limit can be configured via the API using the `WithCardinalityLimit`
+// option.
+//
+// When the configured limit is reached, any new attribute combination is not
+// tracked separately. Instead, its measurements are aggregated together under
+// a special attribute set containing `{"otel.metric.overflow": true}`.
+// This ensures total metric values (e.g., Sum, Count) remain correct for the
+// collection interval, but information about the specific dropped combinations
+// is not preserved.
+//
+// Recommendations:
```

Comment:
Should we document that the OpenTelemetry Specification recommends to use 2000 by default?